libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h

```diff
@@ -205,8 +198,112 @@ class WiFiClientSecure : public WiFiClient {
+    bool stop(unsigned int maxWaitMs) { return _ctx->stop(maxWaitMs); }
+    void flush() override { (void)flush(0); }
+    void stop() override { (void)stop(0); }
+
+    // Allow sessions to be saved/restored automatically to a memory area
+    void setSession(Session *session) { _ctx->setSession(session); }
+
+    // Don't validate the chain, just accept whatever is given.  VERY INSECURE!
+    void setInsecure() { _ctx->setInsecure(); }
+
+    // Assume a given public key, don't validate or use cert info at all
+    void setKnownKey(const PublicKey *pk, unsigned usages = BR_KEYTYPE_KEYX | BR_KEYTYPE_SIGN) {
+      _ctx->setKnownKey(pk, usages);
+    }
+    // Only check SHA1 fingerprint of certificate
+    bool setFingerprint(const uint8_t fingerprint[20]) {
+      return _ctx->setFingerprint(fingerprint);
+    }
+    bool setFingerprint(const char *fpStr) { return _ctx->setFingerprint(fpStr); }
+    // Accept any certificate that's self-signed
+    void allowSelfSignedCerts() { _ctx->allowSelfSignedCerts(); }
+
+    // Install certificates of trusted CAs or specific site
+    void setTrustAnchors(const X509List *ta) { _ctx->setTrustAnchors(ta); }
+    // In cases when NTP is not used, app must set a time manually to check cert validity
+    void setX509Time(time_t now) { _ctx->setX509Time(now); }
+    // Install a client certificate for this connection, in case the server requires it (i.e. MQTT)
+    void setClientRSACert(const X509List *cert, const PrivateKey *sk) { _ctx->setClientRSACert(cert, sk); }
+    void setClientECCert(const X509List *cert, const PrivateKey *sk,
+                         unsigned allowed_usages, unsigned cert_issuer_key_type) {
+      _ctx->setClientECCert(cert, sk, allowed_usages, cert_issuer_key_type);
+    }
+
+    // Sets the requested buffer size for transmit and receive
+    void setBufferSizes(int recv, int xmit) { _ctx->setBufferSizes(recv, xmit); }
+
+    // Returns whether MFLN negotiation for the above buffer sizes succeeded (after connection)
+    int getMFLNStatus() { return _ctx->getMFLNStatus(); }
+
+    // Return an error code and possibly a text string in a passed-in buffer with last SSL failure
+    int getLastSSLError(char *dest = NULL, size_t len = 0) { return _ctx->getLastSSLError(dest, len); }
+
+    // Attach a preconfigured certificate store
+    void setCertStore(CertStore *certStore) { _ctx->setCertStore(certStore); }
+
+    // Select specific ciphers (i.e. optimize for speed over security)
+    // These may be in PROGMEM or RAM, either will run properly
+    bool setCiphers(const uint16_t *cipherAry, int cipherCount) { return _ctx->setCiphers(cipherAry, cipherCount); }
+    bool setCiphers(const std::vector<uint16_t> list) { return _ctx->setCiphers(list); }
+    bool setCiphersLessSecure() { return _ctx->setCiphersLessSecure(); } // Only use the limited set of RSA ciphers without EC
+
+    // Check for Maximum Fragment Length support for given len before connection (possibly insecure)
```

Comment:
FWIW, these don't need to be moved to ctx, they're static and not part of any connection.

Reply:
Should I make `_cipher_list` static in `WiFiClientSecure` ?

Reply:
Actually, this was intended to be a comment about the MFLN calls, not setCipher.  They are not connection oriented, they're static members that just run before a context would be created.

So, leave cipher in and move the probeMFLNs back to the WiFiClientSecure is what I meant.